sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java

```diff
@@ -933,25 +941,35 @@ public Query apply(Function parent, Function inner, Context context) {
                         break;
                     case LteOperator.NAME:
                         includeUpper = true;
+                        includeLower = true;
                         to = distance;
                         break;
                     case LtOperator.NAME:
                         to = distance;
+                        includeLower = true;
                         break;
                     case GteOperator.NAME:
                         from = distance;
                         includeLower = true;
+                        includeUpper = true;
                         break;
                     case GtOperator.NAME:
                         from = distance;
+                        includeUpper = true;
                         break;
                     default:
                         // invalid operator? give up
                         return null;
                 }
-                GeoPoint geoPoint = new GeoPoint(lat, lon);
 
-                return new GeoDistanceRangeQuery(
+                final Version indexCreated = Version.indexCreated(context.indexCache.indexSettings());
+                if (indexCreated.onOrAfter(Version.V_2_2_0)) {
```

Comment:
please do all checks against V_2_3 and handle 2_2 as if it was an earlier version.
We never shipped Crate with 2.2 and don't want to create tests for 2.2 indices